spec/views/layouts/signup_modal.html.erb_spec.rb

```diff
@@ -1,8 +1,17 @@
 require "rails_helper"
 
 RSpec.describe "layouts/_signup_modal.html.erb", type: :view do
-  it "has the tagline" do
+  let(:tagline_text) { "the best community" }
+
+  it "renders the tagline if it is set" do
+    allow(SiteConfig).to receive(:tagline).and_return(tagline_text)
     render
     expect(rendered).to have_text(SiteConfig.tagline)
   end
+
+  it "does not render the tagline if it is not set" do
+    allow(SiteConfig).to receive(:tagline).and_return(nil)
+    render
+    expect(rendered).to have_no_text(tagline_text)
```

Comment:
**nitpick (non-blocking):** We use the `expect().not_to have_text()` form 8 times throughout our tests, but we never use `to have_no_text`. RSpec is a bit smarter than Minitest here and the two forms are equivalent (example uses `have_xpath`, but same consideration applies):

> Be aware that because of this behaviour, the following two statements are not equivalent, and you should always use the latter!
> 
> !page.has_xpath?('a')
> page.has_no_xpath?('a')
> 
> The former would immediately fail because the content has not yet been removed. Only the latter would wait for the asynchronous process to remove the content from the page.
> 
> Capybara's RSpec matchers, however, are smart enough to handle either form. The two following statements are functionally equivalent:
> 
> expect(page).not_to have_xpath('a')
> expect(page).to have_no_xpath('a')

https://www.rubydoc.info/gems/capybara/3.3.1

Reply:
@citizen428 that's fair! i think i only used `to have_no_text` because rspec suggested that, but i agree with you that we ought to keep it consistent. thanks for pointing that out, i'll update the spec to use this the more consistent method here 👍